libcontainer/specconv/spec_linux.go

```diff
@@ -760,13 +763,15 @@ func setupUserNamespace(spec *specs.Spec, config *configs.Config) error {
 }
 
 // parseMountOptions parses the string and returns the flags, propagation
-// flags and any mount data that it contains.
-func parseMountOptions(options []string) (int, []int, string, int) {
+// flags, any mount data that it contains, and {recAttrSet, recAttrClr, extFlags}.
+func parseMountOptions(options []string) (int, []int, string, uint64, uint64, int) {
 	var (
-		flag     int
-		pgflag   []int
-		data     []string
-		extFlags int
+		flag       int
+		pgflag     []int
+		data       []string
+		recAttrSet uint64
+		recAttrClr uint64
+		extFlags   int
```

Comment:
While at it, can we used named return? It will (mostly) eliminate the need for the `var` block here, and the function declaration will be more descriptive.

IOW
```go
// parseMountOptions parses the option string into separate fields
// used by configs.Mount.
func parseMountOptions(options []string) (flags int, propagationFlags []int, data string, recAtterSet, retAttrClr uint64, extensionFlags int) {
    dataItems []string
    ...
    data = strings.Join(dataItems, ",")
    return
}
```

Ideally do that is a separate PR -- then adding recAtttSet/Clr is easy.

Reply:
I had something like this in the works for some time, and I added a few more commits, which can be used as a base for this. PTAL https://github.com/opencontainers/runc/pull/3281 @AkihiroSuda

Reply:
updated